terraform/modules/pre-auth-lambda/iam.tf

```diff
@@ -0,0 +1,51 @@
+resource aws_iam_role role_for_lambda_user_expiry {
+  name               = "Role-Lambda-User-Expiry"
+  assume_role_policy = data.aws_iam_policy_document.policy_assume_role_lambda_user_expiry.json
+}
+
+data aws_iam_policy_document policy_assume_role_lambda_user_expiry {
+  statement {
+    actions = [
+      "sts:AssumeRole",
+    ]
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+resource aws_iam_role_policy role_policy_dynamodb_read_write_user_expiry {
+  name = "Policy-DynamoDB-Table-User-Read-Write"
+  role   = aws_iam_role.role_for_lambda_user_expiry.id
+  policy = data.aws_iam_policy_document.policy_dynamodb_read_write_user_expiry.json
+}
+
+data aws_iam_policy_document policy_dynamodb_read_write_user_expiry {
+  statement {
+    actions = [
+        "dynamodb:PutItem",
+        "dynamodb:GetItem",
+        "dynamodb:Query",
+        "dynamodb:UpdateItem"
+    ]
+    resources = ["${aws_dynamodb_table.dynamodb_table_user.arn}"]
+  }
+}
+
+resource aws_iam_role_policy role_policy_logs_user_expiry {
+  name = "Role-Policy-Logs-User-Expiry"
+  role   = aws_iam_role.role_for_lambda_user_expiry.id
+  policy = data.aws_iam_policy_document.policy_logs_user_expiry.json
+}
+
+data aws_iam_policy_document policy_logs_user_expiry {
+  statement {
+    actions = [
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+    ]
+    resources = ["arn:aws:logs:*:*:*"]
```

Comment:
it should only have permission to write to it's own log group

Reply:
This is done.